File: src/tokens/operator_tokens.zig
const standard: type = @import("std");

// Operator Token Defintion
pub const OperatorToken: type = []const u8;

// Operator Tokens
pub const ADDITION_TOKEN: [1]OperatorToken = "+";
pub const ADDITION_ASSIGNMENT_TOKEN: [2]OperatorToken = "+=";
pub const AND_TOKEN: [2]OperatorToken = "&&";
pub const ASSIGNMENT_TOKEN: [1]OperatorToken = "=";
pub const DIVISION_TOKEN: [1]OperatorToken = "/";
pub const DIVISION_ASSIGNMENT_TOKEN: [2]OperatorToken = "/=";
pub const EQUALITY_TOKEN: [2]OperatorToken = "==";
pub const GREATER_THAN_TOKEN: [1]OperatorToken = ">";
pub const GREATER_OR_EQUAL_TOKEN: [2]OperatorToken = ">=";
pub const LESS_THAN_TOKEN: [1]OperatorToken = "<";
pub const LESS_OR_EQUAL_TOKEN: [2]OperatorToken = "<=";
pub const MODULUS_TOKEN: [1]OperatorToken = "%";
pub const MODULUS_ASSIGNMENT_TOKEN: [2]OperatorToken = "%=";
pub const MULTIPLICATION_TOKEN: [1]OperatorToken = "*";
pub const MULTIPLICATION_ASSIGNMENT_TOKEN: [2]OperatorToken = "*=";
pub const NOT_TOKEN: [1]OperatorToken = "!";
pub const NOT_EQUAL_TOKEN: [2]OperatorToken = "!=";
pub const NULL_COALESCING_TOKEN: [1]OperatorToken = "?";
pub const OR_TOKEN: [2]OperatorToken = "||";
pub const SUBTRACTION_TOKEN: [1]OperatorToken = "-";
pub const SUBTRACTION_ASSIGNMENT_TOKEN: [2]OperatorToken = "-=";
pub const RANGE_TOKEN: [2]OperatorToken = "..";

// Operator Tokens Vector
pub fn operators_vector() @Vector(22, OperatorToken) {
    const operators: @Vector(22, OperatorToken) = @Vector(22, OperatorToken){
        ADDITION_TOKEN,
        ADDITION_ASSIGNMENT_TOKEN,
        AND_TOKEN,
        ASSIGNMENT_TOKEN,
        DIVISION_TOKEN,
        DIVISION_ASSIGNMENT_TOKEN,
        EQUALITY_TOKEN,
        GREATER_THAN_TOKEN,
        GREATER_OR_EQUAL_TOKEN,
        LESS_THAN_TOKEN,
        LESS_OR_EQUAL_TOKEN,
        MODULUS_TOKEN,
        MODULUS_ASSIGNMENT_TOKEN,
        MULTIPLICATION_TOKEN,
        MULTIPLICATION_ASSIGNMENT_TOKEN,
        NOT_TOKEN,
        NOT_EQUAL_TOKEN,
        NULL_COALESCING_TOKEN,
        OR_TOKEN,
        SUBTRACTION_TOKEN,
        SUBTRACTION_ASSIGNMENT_TOKEN,
        RANGE_TOKEN,
    };

    return operators;
}

File: src/tokens/operator_tokens.cc
#include <vector>
#include "./operator_tokens.hh"

namespace tokens {
    using namespace std;
    
    // Operator Token Vector
    vector<OperatorToken> operators_vector() {
        const vector<OperatorToken> operators = {
            ADDITION_TOKEN,
            ADDITION_ASSIGNMENT_TOKEN,
            AND_TOKEN,
            ANNOTATION_TOKEN,
            ASSIGNMENT_TOKEN,
            BITWISE_SHIFT_LEFT_TOKEN,
            BITWISE_SHIFT_RIGHT_TOKEN,
            BITWISE_SHIFT_LEFT_ASSIGNMENT_TOKEN,
            BITWISE_SHIFT_RIGHT_ASSIGNMENT_TOKEN,
            COLON_TOKEN,
            COMMA_TOKEN,
            DIVISION_TOKEN,
            DIVISION_ASSIGNMENT_TOKEN,
            DOLLAR_SIGN_TOKEN,
            EQUALITY_TOKEN,
            EXCLUSIVE_OR_TOKEN,
            FIELD_ACCESS_TOKEN,
            GRAVE_ACCENT_TOKEN,
            GREATER_THAN_TOKEN,
            GREATER_OR_EQUAL_TOKEN,
            INCLUSIVE_RANGE_TOKEN,
            LAMBDA_TOKEN,
            LEFT_ARROW_TOKEN,
            LESS_THAN_TOKEN,
            LESS_OR_EQUAL_TOKEN,
            MODULUS_TOKEN,
            MODULUS_ASSIGNMENT_TOKEN,
            MULTIPLICATION_TOKEN,
            MULTIPLICATION_ASSIGNMENT_TOKEN,
            NOT_TOKEN,
            NOT_EQUAL_TOKEN,
            NULL_COALESCING_TOKEN,
            OR_TOKEN,
            PATH_SEPARATOR_TOKEN,
            RANGE_TOKEN,
            RETURN_TYPE_TOKEN,
            SEMI_COLON_TOKEN,
            SUBTRACTION_TOKEN,
            SUBTRACTION_ASSIGNMENT_TOKEN,
            TILDE_TOKEN,
            UNDERSCORE_TOKEN,
            VARIADIC_RANGE_TOKEN,
        };
    
        return operators;
    }
}